eng/tools/spec-gen-sdk-runner/src/commands.ts

```diff
@@ -325,3 +332,65 @@ function getSpecPaths(runMode: string, specRepoPath: string): string[] {
   }
   return specConfigPaths;
 }
+
+/**
+ * Extract and format the prefix from tspConfigPath or readmePath.
+ * @param {string | undefined} tspConfigPath The tspConfigPath to extract the prefix from.
+ * @param {string | undefined} readmePath The readmePath to extract the prefix from.
+ * @returns {string} The formatted prefix.
+ */
+function extractPathFromSpecConfig(tspConfigPath: string | undefined, readmePath: string | undefined): string {
```

Comment:
Seems we still need this logic to find the log file name in case of batch run.  Can you add a comment saying "this function is copied from 'spec-gen-sdk'" and add the source code link?